cirq-web/cirq_ts/README.md

```diff
@@ -0,0 +1,96 @@
+
+For bundling the Typescript into Javascript that can be run in the browser, and for overall ease of development, we use [Webpack](https://webpack.js.org/).
+
+As an additional note, all `npm` commands can be ran from the top-level of Cirq like so:
+```
+npm --prefix 'cirq-web/cirq_ts' [YOUR_COMMAND]
+```
+### Visualization build structure
+
+The reference example for the build structure of a visualization is the Bloch sphere. Reference the `src/bloch_sphere/main.ts` file and the `src/bloch_sphere/bloch_sphere.ts` file to see the code. The `src/bloch_sphere/` directory should serve as a guide for how Typescript visualizations in Cirq should be structured. Visualizations should have:
+ - A "root" folder within the `src/` directory labeled according to the title of the visualization. All files and directories for a particular visualization will live here. In the case of the Bloch Sphere, this is `bloch_sphere/`.
+ - A `components/` directory which contains classes representing different components of the larger visualization, following typical object oriented programming techniques. In the case of the Bloch sphere, you can see that we have different classes for `Axes`, `Meridians`, `Text` etc.
+ - Any `assets/` directory with information necessary for the visualization (fonts, images, etc.). In the case of the Bloch Sphere, we can see a `fonts/` subdirectory which holds necessary font data, within the `assets/` directory, but for instances where there isn't a lot of extra information necessary subdirectories may not be needed.
+ - A class within the visualization's "root" folder which brings the individual components of the visualization together. In the case of the Bloch Sphere, this is `BlochSphere.class.ts`.
+ - A `main.ts` consisting of functions which will be called from the bundled library. These function should handle:
+    - Receiving any input data which could affect the visualization.
+    - Sending final visualization output to the development environment, notebook, or HTML files. 
+    - Combining aspects of the visualization that need to be added separately.
+
+This `main.ts` file will also need to be added as an entry point in the `webpack.config.js` file in order for your visualization to be bundled accordingly.
+```javascript
+module.exports = {
+  entry: {
+    bloch_sphere: './src/bloch_sphere/main.ts',
+    ...
+    YOUR_VIZ_NAME: './src/YOUR_VIZ_NAME/main.ts',
+  },
+  ...
+};
+```
+You can learn more about Webpack entry points here: [Webpack Entry Points](https://webpack.js.org/concepts/entry-points/)
+
+### Creating visualization bundle files
+
+Following this structure, you will be able to bundle your visualization by running the command `npx webpack --mode production`. This will build the bundled Javascript file(s) within the `dist/` directory, where you can access and reference them in HTML.
```

Comment:
I guess we'll need a `check/npx` as well

Reply:
Right now I go into the `cirq_ts` directory and run, I don't think `--prefix` is supported yet for it, once I'm done i'll do some more digging

https://github.com/npm/npx/issues/74
https://stackoverflow.com/questions/60739550/how-to-specify-pathlike-npm-prefix-in-npx

Reply:
OK, in that case you can simulate that with 
```
#!/bin/bash

cd cirq-web/cirq_ts
npx $@
```